R/death_functions.R

```diff
@@ -960,6 +1000,503 @@ death_injury_matrix_count <- function(ph.data,
+             "which is required for King County data.")
+      }
+
+      if (kingco) {
+        ph.data <- ph.data[chi_geo_kc == 'King County']
+      }
+
+    # group_by ----
+      if (!is.null(group_by)) {
+        group_col_error <- setdiff(group_by, names(ph.data))
+        if (length(group_col_error) > 0) {
+          stop("\n\U0001f6d1 The following `group_by` values are not column names in `ph.data`: ",
+               paste0(group_col_error, collapse = ', '), ".")
+        }
+      }
+
+    # ypll_age ----
+      if (!is.null(ypll_age)) {
+        if (!is.numeric(ypll_age) ||
+            !all(ypll_age == floor(ypll_age)) ||
+            length(ypll_age) != 1 || ypll_age < 1 | ypll_age > 99) {
+          stop("\n\U0001f47f `ypll_age` must be an integer between 1 and 99.")
+        }
+
+        ypll_col_name <- paste0("ypll_", ypll_age)
+      }
+
+    # death_age_col ----
+      if (!is.null(death_age_col)) {
+        if (!(death_age_col %in% names(ph.data))) {
+          stop("\n\U0001f47f `death_age_col` must be the name of column that exists in `ph.data`.")
+        }
+
+        col_data <- unique(ph.data[[death_age_col]])
+        if (!is.numeric(col_data) || !all(is.na(col_data) | col_data == floor(col_data), na.rm = TRUE)) {
+          stop("\n\U0001f47f If `death_age_col` is specified, it must be a column of integers in `ph.data`.")
+        }
+
+        if (is.null(ypll_age)) {
+          stop("\n\U0001f47f `death_age_col` should not be specified when `ypll_age` is NULL.")
+        }
+      }
+
+      if (is.null(death_age_col) & !is.null(ypll_age)) {
+        if ("chi_age" %in% names(ph.data)) {
+          death_age_col <- 'chi_age'
+          message("\U0001F4E3 \nYou requested the calculation of YPLL by specifying `ypll_age` and did not provide `death_age_col`.",
+                  "\nThe function found and used a column named `chi_age` for the YPLL calculation. If this was not",
+                  "\nyour intention, please specify the correct column with the decedent's age with the",
+                  "\n`death_age_col` argument.")
+        } else {
+          stop(paste0("\n\U0001f47f You requested the calculation of YPLL by specifying `ypll_age` and did not provide `death_age_col`.",
+                      "\nThe function attempted to use a column named `chi_age`, but it was not found.",
+                      "\nTo calculate YPLL, please set death_age_col to the name of the column with the age at death."))
+        }
+      }
+
+  # Identify deaths meeting criteria ----
+    # Check underlying cause
+    ph.data[, meets_criteria := icd10_tempy %in% underlying_codes]
+
+    # Check contributing causes (made my brain hurt, but it works!)
+    if (contributing_logic == "ANY") {
+      # ANY logic - at least one contributing code must be present
+      ph.data[meets_criteria == TRUE,
+              meets_criteria := Reduce(`|`, lapply(.SD, function(x) x %in% contributing_codes)),
```

Comment:
you probably could have done `rowSums(.SD %in% contributing_codes)>=1` or similar and gotten the same result.